commit 8c203dbb78ca7a9aed4e2570c866b0f43c752e41
Author: Borislav Petkov <bp@suse.de>
Date:   Tue Nov 1 13:04:41 2016 +0100

    x86/RAS: Add TSC timestamp to the injected MCE
    
    The MCE injection code does not provide the time stamp information for the
    injected MCE. Add it.
    
    Signed-off-by: Borislav Petkov <bp@suse.de>
    Link: http://lkml.kernel.org/r/20161101120911.13163-3-bp@alien8.de
    Signed-off-by: Thomas Gleixner <tglx@linutronix.de>

diff --git a/arch/x86/ras/mce_amd_inj.c b/arch/x86/ras/mce_amd_inj.c
index 1ac76479c266..8730c2882fff 100644
--- a/arch/x86/ras/mce_amd_inj.c
+++ b/arch/x86/ras/mce_amd_inj.c
@@ -275,6 +275,8 @@ static void do_inject(void)
 	unsigned int cpu = i_mce.extcpu;
 	u8 b = i_mce.bank;
 
+	rdtscll(i_mce.tsc);
+
 	if (i_mce.misc)
 		i_mce.status |= MCI_STATUS_MISCV;